tests/unit/CoreListTest.php

```diff
@@ -0,0 +1,373 @@
+		return '
+			fragment CoreListBlockFragment on CoreList {
+				attributes {
+					anchor
+					backgroundColor
+					className
+					cssClassName
+					fontFamily
+					fontSize
+					gradient
+					lock
+					# metadata
+					ordered
+					placeholder
+					reversed
+					start
+					style
+					textColor
+					type
+					values
+				}
+			}
+
+			query Post( $id: ID! ) {
+				post(id: $id, idType: DATABASE_ID) {
+					databaseId
+					editorBlocks {
+						apiVersion
+						blockEditorCategoryName
+						clientId
+						cssClassNames
+						innerBlocks {
+							name
+						}
+						isDynamic
+						name
+						parentClientId
+						renderedHtml
+						...CoreListBlockFragment
+					}
+				}
+			}
+		';
+	}
+
+	/**
+	 * Test case for retrieving core list block fields and attributes.
+	 *
+	 * The following aspects are tested:
+	 * - The absence of errors in the GraphQL response.
+	 * - Presence of the 'data' and 'post' keys in the response.
+	 * - Matching post ID.
+	 * - Correct block name ('core/list').
+	 * - Correct retrieval of the block's attributes, especially 'anchor', 'backgroundColor', 'className', and 'cssClassName'.
+	 *
+	 * @return void
+	 */
+	public function test_retrieve_core_list_fields_and_attribute() {
+		$block_content = '
+			<!-- wp:list {"className":"test-css-class-name","backgroundColor":"accent-4"} -->
+				<ul id="test-anchor" class="wp-block-list test-css-class-name has-accent-4-background-color has-background">
+					<!-- wp:list-item --><li>Truck</li><!-- /wp:list-item -->
+					<!-- wp:list-item --><li>Train</li><!-- /wp:list-item -->
+				</ul>
+			<!-- /wp:list -->
+		';
+
+		// Set post content.
+		wp_update_post(
+			[
+				'ID'           => $this->post_id,
+				'post_content' => $block_content,
+			]
+		);
+
+		$query     = $this->query();
+		$variables = [
+			'id' => $this->post_id,
+		];
+
+		$actual = graphql( compact( 'query', 'variables' ) );
+
+		$this->assertArrayNotHasKey( 'errors', $actual, 'There should not be any errors' );
+		$this->assertArrayHasKey( 'data', $actual, 'The data key should be present' );
+		$this->assertArrayHasKey( 'post', $actual['data'], 'The post key should be present' );
+		$this->assertEquals( $this->post_id, $actual['data']['post']['databaseId'], 'The post ID should match' );
+
+		$block = $actual['data']['post']['editorBlocks'][0];
+		$this->assertEquals( 'core/list', $block['name'], 'The block name should be core/list' );
+
+		$this->assertEquals(
+			[
+				'anchor'          => 'test-anchor',
+				'backgroundColor' => 'accent-4',
+				'className'       => 'test-css-class-name',
+				'cssClassName'    => 'wp-block-list test-css-class-name has-accent-4-background-color has-background',
+				'fontFamily'      => null,
+				'fontSize'        => null,
+				'gradient'        => null,
+				'lock'            => null,
+				'ordered'         => false,
+				'placeholder'     => null, // @todo : Untested as it is getting returned as null.
+				'reversed'        => null,
```

Comment:
This is redundant, of course its `null` if it's untested. This can just be noted in the PR description (like `lock`)